src/components/ExchangePage/index.jsx

```diff
@@ -235,14 +242,42 @@ function getMarketRate(
   }
 }
 
-export default function ExchangePage({ initialCurrency, sending }) {
+export default function ExchangePage({ initialCurrency, sending = false, params }) {
   const { t } = useTranslation()
   const { account } = useWeb3Context()
 
+  const [, toggleDarkMode] = useDarkModeManager()
+
+  useEffect(() => {
+    if (params.theme) {
+      toggleDarkMode(
+        params.theme === SUPPORTED_THEMES.DARK ? true : params.theme === SUPPORTED_THEMES.LIGHT ? false : ''
+      )
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
in general, disabling `eslint-disable-line react-hooks/exhaustive-deps` is pretty scary, in this case though i think we can allow it because we only care about toggling the theme on mount.

however, we should validate that `params.theme` is equal to one of the supported themes, rather than just passing `''` to `toggleDarkMode` if it's not.